apisix/admin/standalone.lua

```diff
@@ -193,8 +206,28 @@ function _M.run()
     end
 end
 
+local function cleanup_on_exit()
+    if ngx.worker.exiting() then
+        local status_shdict = ngx.shared.status_report_standalone
+        local pid = tostring(ngx.worker.pid())
+        status_shdict:delete(pid)
+        core.log.info("worker ", pid, " removed itself on exit")
+        return
+    end
+
+    -- Reschedule the timer to check again
+    local ok, err = ngx.timer.at(0.1, cleanup_on_exit)
```

Comment:
Using a timer to periodically check whether the shared dict needs cleaning feels wasteful. 
Can we change the records in this shared dict to **a list of worker PIDs waiting for synchronization configuration**? When `init_worker` is completed, add the current worker's PID to this list. When `update_config` is completed, directly remove this PID from the list, thus eliminating the need for subsequent checks.

Reply:
In the case when worker exits(like reload), it wont be removed from the list causing health check to fail indefinitely.

Reply:
got it.